packages/app/src/components/Root.tsx

```diff
@@ -0,0 +1,82 @@
+import { UpsetActions, UpsetProvenance } from "@visdesignlab/upset2-react"
+import { UpsetConfig } from "@visdesignlab/upset2-core"
+import { Box, css } from "@mui/material"
+import { Body } from "./Body"
+import Header from "./Header"
+import { useRef, useState, useEffect, createContext } from "react"
+import React from "react"
+
+type Props = {
+    provenance: UpsetProvenance,
+    actions: UpsetActions,
+    data: any,
+    config?: UpsetConfig
+}
+
+const AppCss = css`
+  overflow: hidden;
+  height: 100vh;
+  display: grid;
+  grid-template-rows: min-content auto;
+`;
+
+export const ProvenanceContext = createContext<{
+    provenance: UpsetProvenance;
+    actions: UpsetActions;
+    isAtLatest: boolean;
+    isAtRoot: boolean;
+  }>(undefined!);
+
+export const Root = ({provenance, actions, data, config}: Props) => {
+    const ref = useRef<HTMLDivElement>(null);
```

Comment:
This is a bad variable name — it's not descriptive enough. Can we update this to make it a bit more clear?